Source/WebCore/css/CSSCustomPropertyValue.h

```diff
@@ -126,12 +73,50 @@ class CSSCustomPropertyValue final : public CSSValue {
     {
     }
 
+    const Vector<CSSParserToken>& tokens() const;
+
     const AtomString m_name;
     const VariantValue m_value;
     mutable String m_cachedCSSText;
-    mutable RefPtr<CSSVariableData> m_cachedTokens;
 };
 
+inline Ref<CSSCustomPropertyValue> CSSCustomPropertyValue::createUnresolved(const AtomString& name, Ref<CSSVariableReferenceValue>&& value)
+{
+    return adoptRef(*new CSSCustomPropertyValue(name, VariantValue { WTF::InPlaceType<Ref<CSSVariableReferenceValue>>, WTFMove(value) }));
+}
+
+inline Ref<CSSCustomPropertyValue> CSSCustomPropertyValue::createSyntaxAll(const AtomString& name, Ref<CSSVariableData>&& value)
+{
+    return adoptRef(*new CSSCustomPropertyValue(name, VariantValue { WTF::InPlaceType<Ref<CSSVariableData>>, WTFMove(value) }));
+}
+
+inline Ref<CSSCustomPropertyValue> CSSCustomPropertyValue::createWithCSSWideKeyword(const AtomString& name, CSSWideKeyword keyword)
+{
+    return adoptRef(*new CSSCustomPropertyValue(name, VariantValue { keyword }));
+}
+
+inline bool CSSCustomPropertyValue::isVariableReference() const
+{
+    return std::holds_alternative<Ref<CSSVariableReferenceValue>>(m_value);
+}
+
+inline bool CSSCustomPropertyValue::isVariableData() const
+{
+    return std::holds_alternative<Ref<CSSVariableData>>(m_value);
+}
+
+inline bool CSSCustomPropertyValue::isCSSWideKeyword() const
+{
+    return std::holds_alternative<CSSWideKeyword>(m_value);
+}
+
+inline std::optional<CSSWideKeyword> CSSCustomPropertyValue::tryCSSWideKeyword() const
+{
+    if (auto* keyword = std::get_if<CSSWideKeyword>(&m_value))
+        return *keyword;
+    return { };
+}
```

Comment:
Can these be in the `.cpp` file and still get sufficiently inlined and collapsed due to link-time optimization, or is it important to put them in the header?